gobblin-config-management/gobblin-config-client/src/main/java/gobblin/config/client/ConfigClientUtils.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (C) 2015 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.config.client;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.Collection;
+
+import com.google.common.base.Preconditions;
+
+import gobblin.config.common.impl.SingleLinkedListConfigKeyPath;
+import gobblin.config.store.api.ConfigKeyPath;
+import gobblin.config.store.api.ConfigStore;
+
+public class ConfigClientUtils {
+
+  /**
+   * 
+   * @param configKeyURI - URI provided by client , which could missing authority/store root directory
+   * @param cs           - ConfigStore corresponding to the input URI. Require input URI's scheme/authority name 
+   *                       match ConfigStore's scheme/authority
+   * @return             - {@link ConfigKeyPath} for the relative path
+   */
+  public static ConfigKeyPath buildConfigKeyPath(URI configKeyURI, ConfigStore cs){
+    preconditionCheck(configKeyURI, cs);
+    // Example store root is   etl-hdfs://eat1-nertznn01.grid.linkedin.com:9000/user/mitu/HdfsBasedConfigTest
+    
+    // configKeyURI is etl-hdfs:///datasets/a1/a2
+    if(configKeyURI.getAuthority()==null){
+      return getConfigKeyPath(configKeyURI.getPath());
+    }
+    // configKeyURI is etl-hdfs://eat1-nertznn01.grid.linkedin.com:9000/user/mitu/HdfsBasedConfigTest/datasets/a1/a2
+    else {
+      URI relative = cs.getStoreURI().relativize(configKeyURI);
+      return getConfigKeyPath(relative.getPath());
+    } 
+  }
+  
+  /**
+   * Require input URI's scheme/authority name match ConfigStore's scheme/authority
+   * if the input URI's authority is present, return the input ConfigStore's root URI
+   * else, return the URI which only contains the scheme name
+   * 
+   * @param configKeyURI
+   * @param cs
+   * @return
+   */
+  public static URI getDefaultRootURI(URI configKeyURI, ConfigStore cs){
+    preconditionCheck(configKeyURI, cs);
+    
+    if(configKeyURI.getAuthority()!=null){
+      return cs.getStoreURI();
```

Comment:
Isn't the assumption that the authority is missing? Shouldn't this error out instead?